### PR TITLE
gh-101907: Stop using `_Py_OPCODE` and `_Py_OPARG` macros

### PR DESCRIPTION
**Pull request by zooba**
Stops using the macros throughout our core codebase and makes them compatible with previous releases.

* Issue: gh-101907

### CODE REVIEW

**Comment by zooba** [comment]
Posted as a draft because it doesn't build for some reason deep inside macros in generated code in ceval. I'll try to figure it out later

File: Include/cpython/code.h [hunk 1]
```diff
@@ -19,21 +19,35 @@ extern "C" {
...
+    word.op.arg = oparg;
+    return word.cache;
+}
 
 static inline void
-_py_set_opcode(_Py_CODEUNIT *word, uint8_t opcode)
+_py_set_opcode(_Py_CODEUNIT *word, uint16_t packed_opcode)
 {
-    word->opcode = opcode;
+    word->cache = packed_opcode;
```
Comment (markshannon) [inline Include/cpython/code.h]: This appears to be wrong.
It should be `word->op.code = opcode`

File: Tools/cases_generator/generate_cases.py [hunk 2]
```diff
@@ -17,7 +18,7 @@
 
 HERE = os.path.dirname(__file__)
 ROOT = os.path.join(HERE, "../..")
-THIS = os.path.relpath(__file__, ROOT)
+THIS = os.path.relpath(__file__, ROOT).replace(os.path.sep, posixpath.sep)
```
Comment (markshannon) [inline Tools/cases_generator/generate_cases.py:21]: I assume this is necessary for WIndows?

Maybe this shouldn't be in a PR named "Stop using _Py_OPCODE and _Py_OPARG macros", but I don't see any harm in it.

**Review by markshannon — CHANGES_REQUESTED** [review]
Thanks for this. It makes the code a bit easy to reason about.

The change to to `_py_set_opcode` looks wrong, but otherwise LGTM.

File: Include/cpython/code.h (hunk 1 above)
Comment (zooba) [reply Include/cpython/code.h]: Yeah, you're right. I thought I had reimplemented the previous versions semantics properly, but turns out I got it wrong.

File: Tools/cases_generator/generate_cases.py (hunk 2 above)
Comment (zooba) [reply Tools/cases_generator/generate_cases.py:21]: It's only necessary to stop the comments at the top of the file reflecting which OS the script was last run on.

If I don't make this change, I have to manually fix the regenerated files. No big deal either way, but it was less work to fix the generator.

**Comment by zooba** [comment]
I have made the requested changes; please review again

**Comment by markshannon** [comment]
LGTM.

Thanks for doing this.

**Pull request merged**